apps/web/src/components/workflow/preview/push/Content.tsx

```diff
@@ -0,0 +1,185 @@
+import { colors, Text } from '@novu/design-system';
+import { useEffect, useState } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { useGetLocalesFromContent, usePreviewPush } from '../../../../api/hooks';
+import { IS_DOCKER_HOSTED } from '../../../../config';
+import { useAuthController, useDataRef, useProcessVariables } from '../../../../hooks';
+import { IForm } from '../../../../pages/templates/components/formTypes';
+import { useStepFormCombinedErrors } from '../../../../pages/templates/hooks/useStepFormCombinedErrors';
+import { useStepFormPath } from '../../../../pages/templates/hooks/useStepFormPath';
+import { LocaleSelect, PreviewEditOverlay } from '../common';
+import { NovuGreyIcon } from '../common/NovuGreyIcon';
+import {
+  ContentAndOVerlayWrapperStyled,
+  ContentHeaderStyled,
+  ContentStyled,
+  ContentWrapperStyled,
+} from './Content.styles';
+
+export default function Content() {
+  const [isEditOverlayVisible, setIsEditOverlayVisible] = useState(false);
+  const { colorScheme } = useMantineColorScheme();
+  const isDark = colorScheme === 'dark';
+
+  const { control } = useFormContext<IForm>();
+  const path = useStepFormPath();
+  const errorMsg = useStepFormCombinedErrors();
+
+  const [selectedLocale, setSelectedLocale] = useState<string | undefined>(undefined);
+
+  const [parsedPreviewState, setParsedPreviewState] = useState({
+    title: '',
+    content: '',
+  });
+
+  const { getPushPreview, isLoading } = usePreviewPush({
+    onSuccess: (result) => {
+      setParsedPreviewState({
+        content: result.content,
+        title: result.title,
+      });
+    },
+  });
+
+  const title = useWatch({
+    name: `${path}.template.title`,
+    control,
+  });
+
+  const content = useWatch({
+    name: `${path}.template.content`,
+    control,
+  });
+
+  const variables = useWatch({
+    name: `${path}.template.variables`,
+    control,
+  });
+
+  const { organization } = useAuthController();
+  const processedVariables = useProcessVariables(variables);
+
+  const previewData = useDataRef({ content, title });
+
+  const { data: locales, getLocalesFromContent, isLoading: isLocalesLoading } = useGetLocalesFromContent();
+
+  useEffect(() => {
+    getPushPreview({
+      content: previewData.current.content,
+      title: previewData.current.title,
+      locale: selectedLocale,
+      payload: processedVariables,
+    });
+  }, [getPushPreview, previewData, processedVariables, selectedLocale]);
+
+  useEffect(() => {
+    if (!IS_DOCKER_HOSTED) {
+      /*
+       * combining title and content to get locales based upon variables in both title and content
+       * The api is not concerned about the content type, it will parse the given string and return the locales
+       */
+      const combinedContent = `${previewData.current.title} ${previewData.current.content}`;
+      getLocalesFromContent({
+        content: combinedContent,
+      });
+    }
```

Comment:
The API parses any given string and returns the locales of the translation groups variable used in the given string. So it is okay to combine the title and content string into one to pass it to the API.